### PR TITLE
Eip 900 update

### PR DESCRIPTION
**Pull request by 0xAshish**

### CODE REVIEW

File: contracts/root/StakeManager.sol [hunk 1]
```diff
@@ -32,7 +32,7 @@ contract StakeManager is StakeManagerInterface, RootChainable, Lockable {
   event DynastyValueChange(uint256 newDynasty, uint256 oldDynasty);
 
   // optional event to ack unstaking
-  event UnstakeInit(address indexed user, uint256 indexed amount, bytes data);
+  event UnstakeInit(address indexed user, uint256 indexed amount, uint256 indexed deactivationEpoch, bytes data);
```
Comment (jdkanani) [inline contracts/root/StakeManager.sol]: change `user` to `validator`

File: contracts/root/StakeManager.sol [hunk 2]
```diff
@@ -164,7 +160,7 @@ contract StakeManager is StakeManagerInterface, RootChainable, Lockable {
     validatorState[exitEpoch].stakerCount = (
       validatorState[exitEpoch].stakerCount - 1);
 
-    emit UnstakeInit(msg.sender, amount, abi.encode(exitEpoch));
+    emit UnstakeInit(msg.sender, amount, exitEpoch, "0x0");
```
Comment (jdkanani) [inline contracts/root/StakeManager.sol]: "0x0"? Do we need it?

File: contracts/root/StakeManagerInterface.sol [hunk 3]
```diff
@@ -3,11 +3,11 @@ pragma solidity ^0.4.24;
...
 // ERC900
 interface StakeManagerInterface {
-  event Staked(address indexed user, uint256 amount, uint256 total, bytes data);
+  event Staked(address indexed user, address indexed signer, uint256 indexed activatonEpoch, uint256 amount, uint256 total,  bytes data);
   event Unstaked(address indexed user, uint256 amount, uint256 total, bytes data);
 
-  function stake(uint256 amount, bytes data) public;
-  function stakeFor(address user, uint256 amount, bytes data) public;
+  function stake(address unstakeValidator, address signer, uint256 amount,  bytes data) public;
+  function stakeFor(address user, address unstakeValidator, address signer, uint256 amount, bytes data) public;
```
Comment (jdkanani) [inline contracts/root/StakeManagerInterface.sol]: Do we need `bytes data`?

File: contracts/root/StakeManagerInterface.sol [hunk 4]
```diff
@@ -3,11 +3,11 @@ pragma solidity ^0.4.24;
 
 // ERC900
 interface StakeManagerInterface {
-  event Staked(address indexed user, uint256 amount, uint256 total, bytes data);
+  event Staked(address indexed user, address indexed signer, uint256 indexed activatonEpoch, uint256 amount, uint256 total,  bytes data);
```
Comment (jdkanani) [inline contracts/root/StakeManagerInterface.sol]: Do we need `bytes data`?

File: contracts/root/StakeManagerInterface.sol [hunk 5]
```diff
@@ -3,11 +3,11 @@ pragma solidity ^0.4.24;
...
   event Unstaked(address indexed user, uint256 amount, uint256 total, bytes data);
 
-  function stake(uint256 amount, bytes data) public;
-  function stakeFor(address user, uint256 amount, bytes data) public;
+  function stake(address unstakeValidator, address signer, uint256 amount,  bytes data) public;
```
Comment (jdkanani) [inline contracts/root/StakeManagerInterface.sol]: Do we need `bytes data`?

Comment (vaibhavchellani) [reply contracts/root/StakeManagerInterface.sol]: i think for signer we do need `bytes data`

Comment (0xAshish) [reply contracts/root/StakeManagerInterface.sol]: @vaibhavchellani signer is a param now.

**Comment by jdkanani** [comment]
@0xAshish

File: contracts/root/StakeManagerInterface.sol (hunk 4 above)
Comment (0xAshish) [reply contracts/root/StakeManagerInterface.sol]: not immediately, we might need it in the future.

File: contracts/root/StakeManagerInterface.sol (hunk 5 above)
Comment (vaibhavchellani) [reply contracts/root/StakeManagerInterface.sol]: i thought we cannot add in param due to EIP-900 signature matching

**Pull request merged**